src/main/java/org/jalasoft/moi/model/core/Executer.java

```diff
@@ -9,50 +9,101 @@
 
 package org.jalasoft.moi.model.core;
 
-import java.io.BufferedReader;
+import org.jalasoft.moi.model.core.parameters.InputParameters;
+import org.jalasoft.moi.model.core.parameters.ProcessResult;
+import org.jalasoft.moi.model.core.parameters.Result;
+
+import java.io.BufferedWriter;
 import java.io.IOException;
+import java.io.InputStream;
 import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.util.Objects;
 
 /**
- * Class recieves a string, executes it on cmd and returns output on a string
+ * Class receives a string, executes it on cmd and returns output on a string
  *
  * @author Mauricio Oroza
- * @version 1.0 03 March 2020
+ *         Lucero Quiroga Perez
+ * @version 1.1 03 March 2020
  */
 public class Executer {
 
-    private String command;
+    private ICacheProvider cache;
+    private Result result;
+
+    public Executer(ICacheProvider cache) {
+        this.cache = cache;
+        this.result = new ProcessResult();
+    }
 
     /**
-     * Constructor receives string command
+     * Executes command in cmd
      *
-     * @param command command to be executed
+     * @return The output of the console in one string in the form: String1 + \n + String1 + \n + ...
      */
-    public Executer(String command) {
-        String commandSlash = "\"";
-        this.command = "cmd /c " + commandSlash + command + commandSlash;
+    public Result execute(String command) throws IOException {
+        String builtCommand = "cmd /c \"" + command + "\"";
+        Process tempProcess = Runtime.getRuntime().exec(builtCommand);
+        long pid = getPid(tempProcess.toString());
+        cache.add(pid, tempProcess);
+        result.setPid(pid);
+        result.setValue(buildResult(tempProcess));
+        return result;
     }
 
     /**
-     * Executes command in cmd
+     * Processes an input to continue the execution.
      *
-     * @return The output of the console in one string in the form: String1 + \n + String1 + \n + ...
+     * @param answer user input
+     * @return a result value and the process id
+     * @throws IOException when there is a execution problem
+     */
+    public Result processAnswer(InputParameters answer) throws IOException {
+        Process process = cache.getProcessById(answer.getProcessId());
+        BufferedWriter writer = new BufferedWriter(
+                new OutputStreamWriter(Objects.requireNonNull(process).getOutputStream()));
+        writer.write(answer.getValue() + System.lineSeparator());
+        writer.flush();
```

Comment:
I know it is needed to use flush to keep the writer alive for the rest of the process, but when does it actually close?

Reply:
Because the life cycle of Streams is process controlled, if we interfere with natural flow, an error may occur or the actual process may end.